### PR TITLE
fix(InteractionTag): use color brand foreground2 for brand color instead of brand foreground 1

### LINKED ISSUE #30904: [Bug]: InteractiveTag brand appearance needs better contrast
### Library

React Components / v9 (@fluentui/react-components)

### System Info

```shell
n/a
```

### Are you reporting Accessibility issue?

yes

### Reproduction

https://react.fluentui.dev/?path=/docs/components-tag-interactiontag--default#appearance

## Actual Behavior
1. go to https://react.fluentui.dev/?path=/docs/components-tag-interactiontag--default
2. Change the theme to "Teams Dark"
3. scroll down to _[Appearance](https://react.fluentui.dev/?path=/docs/components-tag-interactiontag--default#appearance)_ section
4. Since button is using Foreground: BrandForeground1 with Background: BrandBackground2, the contrast is 4.04:1
image

## Expected Behavior
Contrast of the InteractiveTag should be 4.5:1. Likely, the foreground was meant to be BrandForeground2 instead of 1. This would be consistent with the "Tag" component rest state for brand, as well as with the Outline appearance, all of which use foreground version "2" of the colors. 
Changing this to BrandForeground2 gives it a 6.38 contrast, and improves all theme's contrast for the brand appearance, too, bringing Web Light's version from 4.8:1 to 5.94:1 for example.
image

### Logs

_No response_

### Requested priority

Normal

### Products/sites affected

_No response_

### Are you willing to submit a PR to fix?

no

### Validations

- [X] Check that there isn't already an issue that reports the same bug to avoid creating a duplicate.
- [X] The provided reproduction is a minimal reproducible example of the bug.

### PR DESCRIPTION
**Pull request by YuanboXue-Amber**
fix #30904 
increase color contrast for brand variant InteractionTag

### CODE REVIEW

**Comment by fabricteam** [comment]
## [Perf Analysis (`@fluentui/react-components`)](https://github.com/microsoft/fluentui/wiki/Perf-Testing)
<p>No significant results to display.</p><details><summary>All results</summary><p>
  <table>
  <tr>
    <th>Scenario</th>
    <th>Render type</th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">Master Ticks</a>
    </th>
    <th>
      <a href="https://github.com/microsoft/fluentui/wiki/Perf-Testing#why-are-results-listed-in-ticks-instead-of-time-units">PR Ticks</a>
    </th>
    <th>Iterations</th>
    <th>Status</th>
  </tr><tr>
            <td>Avatar</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Avatar-mount_base.html">646</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Avatar-mount.html">638</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Button</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Button-mount_base.html">296</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Button-mount.html">298</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>Field</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Field-mount_base.html">1135</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Field-mount.html">1135</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProvider</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProvider-mount_base.html">723</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProvider-mount.html">707</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProviderWithTheme-mount_base.html">75</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProviderWithTheme-mount.html">80</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProviderWithTheme-virtual-rerender_base.html">63</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProviderWithTheme-virtual-rerender.html">63</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>FluentProviderWithTheme</td>
            <td>virtual-rerender-with-unmount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount_base.html">76</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/FluentProviderWithTheme-virtual-rerender-with-unmount.html">76</a></td>
            <td>10</td>
            <td></td>
           </tr>
<tr>
            <td>MakeStyles</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/MakeStyles-mount_base.html">908</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/MakeStyles-mount.html">864</a></td>
            <td>50000</td>
            <td></td>
           </tr>
<tr>
            <td>Persona</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Persona-mount_base.html">1759</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/Persona-mount.html">1753</a></td>
            <td>5000</td>
            <td></td>
           </tr>
<tr>
            <td>SpinButton</td>
            <td>mount</td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/SpinButton-mount_base.html">1417</a></td>
            <td><a href="https://fluentuipr.z22.web.core.windows.net/pull/30973/perf-test-react-components/SpinButton-mount.html">1404</a></td>
            <td>5000</td>
            <td></td>
           </tr></table></p></details>

<div id="perfComment9425"/>

**Comment by fabricteam** [comment]
## 📊 Bundle size report

✅ No changes found
<div id="bundleSizeComment9423"/>

**Review by bsunderhus — APPROVED** [review]

File: change/@fluentui-react-tags-62b2fcad-e929-416b-88a0-1f6617266d80.json [hunk 1]
```diff
@@ -0,0 +1,7 @@
+{
```
Comment (fabricteam) [inline change/@fluentui-react-tags-62b2fcad-e929-416b-88a0-1f6617266d80.json:1]: ## [🕵 fluentuiv9 Open the Visual Regressions report to inspect the affected screenshots]( https://staticpagerelease.z16.web.core.windows.net/azurestaticapps/vr-approval/index.html?organization=uifabric&project=fabricpublic&prId=30973&commitId=255fa28442cec721a15b92124a11745ac6db6600&env=prod&clientType=FLUENTUI)


  <details><summary><b>InteractionTag Converged</b> 8 screenshots</summary>

|Image Name|Diff(in Pixels)| Image Type
|-|-|-|
|InteractionTag Converged.Brand - Dark Mode.hover content.chromium.png|1| Changed|
|InteractionTag Converged.Brand.default.chromium.png|9| Changed|
|InteractionTag Converged.Brand - Dark Mode.pressed dismiss.chromium.png|14| Changed|
|InteractionTag Converged.Brand - Dark Mode.hover dismiss.chromium.png|14| Changed|
|InteractionTag Converged.Brand - Dark Mode.pressed content.chromium.png|1| Changed|
|InteractionTag Converged.Brand - Dark Mode.default.chromium.png|16| Changed|
|InteractionTag Converged.Brand.hover dismiss.chromium.png|9| Changed|
|InteractionTag Converged.Brand.pressed dismiss.chromium.png|9| Changed|

</details>




<div id="vrtCommentfluent-ui_VRT_Pipeline_v9"/>

**Pull request merged**